src/api/v2/quickPickWizard/QuickPickAppResourceStep.ts

```diff
@@ -0,0 +1,94 @@
+import { ApplicationResourceProviderManager } from '../providers/ApplicationResourceProviderManager';
+import { ApplicationResource, ResourceModelBase } from '../v2AzureResourcesApi';
+import { matchesContextValueFilter } from './ContextValueFilter';
+import { QuickPickAppResourceWizardContext } from './QuickPickAppResourceWizardContext';
+import { RecursiveQuickPickStep } from './RecursiveQuickPickStep';
+
+export class QuickPickAppResourceStep<TModel extends ResourceModelBase> extends AzureWizardPromptStep<QuickPickAppResourceWizardContext<TModel>> {
+    public constructor(
+        private readonly resourceProviderManager: ApplicationResourceProviderManager,
+        private readonly branchDataProviderManager: BranchDataProviderManager,
+        private readonly options: PickAppResourceOptions
+    ) {
+        super();
+    }
+
+    public override async prompt(wizardContext: QuickPickAppResourceWizardContext<TModel>): Promise<void> {
+        const allResources = (await this.resourceProviderManager.getResources( /* TODO: subscription */)) || [];
+
+        const matchingResources = allResources.filter(this.matchesAppResource);
+        const picks = matchingResources.map(r => this.getQuickPickItem(r));
+
+        const selected = await wizardContext.ui.showQuickPick(picks, { /* TODO: options */ });
+        wizardContext.applicationResource = selected.data;
+
+        const bdp = this.branchDataProviderManager.getApplicationResourceBranchDataProvider(wizardContext.applicationResource);
+        wizardContext.currentNode = await Promise.resolve(bdp.getResourceItem(wizardContext.applicationResource)) as TModel;
+    }
+
+    public shouldPrompt(_wizardContext: QuickPickAppResourceWizardContext<TModel>): boolean {
+        return true;
+    }
+
+    public async getSubWizard(wizardContext: QuickPickAppResourceWizardContext<TModel>): Promise<IWizardOptions<QuickPickAppResourceWizardContext<TModel>> | undefined> {
+        if (this.options.expectedChildContextValue) {
+            if (matchesContextValueFilter(wizardContext.currentNode as TModel, this.options.expectedChildContextValue)) {
+                return undefined;
+            }
+
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const bdp = this.branchDataProviderManager.getApplicationResourceBranchDataProvider(wizardContext.applicationResource!);
+            return {
+                hideStepCount: true,
+                promptSteps: [
+                    new RecursiveQuickPickStep(bdp, this.options.expectedChildContextValue),
+                ],
+            };
+        }
+
+        return undefined;
+    }
+
+    private getQuickPickItem(resource: ApplicationResource): IAzureQuickPickItem<ApplicationResource> {
+        return {
+            label: resource.name,
+            data: resource,
+        };
+    }
+
+    private matchesAppResource(resource: ApplicationResource): boolean {

```

Comment:
Could we leverage some of the code in [`SubscriptionTreeItem.pickAppResource`](https://github.com/microsoft/vscode-azureresourcegroups/blob/main/src/tree/SubscriptionTreeItem.ts#L37-L83) or maybe extract it out as a util function?

I think the only reason it lives on the Sub treeItem is because it directly accesses the appResources cache, but definitely doesn't need to be that way.

Reply:
I guess that [this ](https://github.com/microsoft/vscode-azuretools/pull/1191/files?diff=split&w=1)is for?

Reply:
Phil's code is different enough that I thought it made the most sense to just write from scratch.